Add section on FeatureServers (#11)
R-ArcGIS/r-bridge-site: location-services/read-data.qmd

  collect()
```

## Map and Feature Servers

> This example has only illustrated how to work with `FeatureLayer` objects. However, often times you may wish to work with a collection of layers in either a `FeatureServer` or `MapServer`. Both of these are collections of multiple layers. Like a `FeatureLayer`, these are accessed with `arc_open()`.

Reviewer: ```suggestion
This example has only illustrated how to work with `FeatureLayer` objects. However, often times you may wish to work with a collection of layers in a `FeatureServer`, `MapServer`, or `GroupLayer`. All of these are collections of multiple layers. Like a `FeatureLayer`, these are accessed with `arc_open()`.
```
I tested and this also works with a GroupLayer?! super cool. It's a little confusing because a GroupLayer is part of a MapServer so it ends up being an extra nested FeatureLayer situation, but hopefully this would be clear enough for people to figure it out?

```{r}
furl <- "https://services3.arcgis.com/ZvidGQkLaDJxRSJ2/arcgis/rest/services/PLACES_LocalData_for_BetterHealth/FeatureServer"

fsrv <- arc_open(furl)
fsrv
```

This `FeatureServer` contains 5 layers. The individual layers can be fetched using `get_layer()` which lets us specify the layer by ID or by name. It is recommended to use the ID as that will be less prone to human error (for example a space is secretly a tab). The result of the function is a `FeatureLayer` object that can be used with `arc_select()` as illustrated above.

```{r}
get_layer(fsrv, id = 2)
```

Some `FeatureServer`s will also contain tables.

```{r}
furl <- "https://services.arcgis.com/P3ePLMYs2RVChkJx/arcgis/rest/services/USA_Wetlands/FeatureServer"
fsrv2 <- arc_open(furl)
fsrv2
```

This can be fetched using `get_layer()` as well.

```{r}
get_layer(fsrv2, 1)
```

If you would like to fetch multiple items at one time there is a plural `get_layers()` which will fetch multiple items based on name or id and return a list.

```{r}
get_layers(fsrv, id = c(0, 2, 4))
```

> There is also a helper `get_all_layers()` to fetch all of layers of a server into a list. The list has two elements `layers` and `tables`. The former containing all of the `FeatureLayer`s and the latter containing all of the `Table`s in the `FeatureServer`.

Reviewer: ```suggestion
There is also a helper `get_all_layers()` to fetch all of layers of a `FeatureServer`, `MapServer`, or `GroupLayer` into a list. The list has two elements `layers` and `tables`. The former containing all of the `FeatureLayer`s and the latter containing all of the `Table`s in the `FeatureServer`.
```
Modifying to add in GroupLayer (which is not a server, I suppose).
